### DISCUSSION TITLE
Can't record rotors state and linear velocity

### DISCUSSION THREAD

**Discussion by fcocina** [microsoft/AirSim#3665] [Support Q&A]
Hi everyone! 
I have a problem in recording some data like rotors state ( thrust, torque , angular_speed) and velocity components (estimated).

I need to record data for all the path of my Multirotor, so I tried getting the state of the Multirotor with a "while " loop, but it seems that there is a problem of getting data while calling API functions like "moveOnPath" , etc. 

Can anyone help me?

**Comment by ahmed-elsaharti** [comment]
Hi @fcocina, I haven't tried this yet but do you get an error message when you try to grab the state while moving on path or does it return an empty list? If it's an error message could you please include the message and/or a code snippet so that we could replicate the problem? ^_^
